doc/ui-guidelines.md

```diff
@@ -0,0 +1,252 @@
+- `:f>` not needed anymore, all components are functional by default
+- `rn/use-callback` hook should be used for anon callback functions
+
+> [!IMPORTANT]
+> DO NOT USE anon functions directly in the props
+ 
+BAD
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         #(set-pressed true)
+      :on-press-out        #(set-pressed nil)}]))
+```
+
+GOOD:
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out}]))
+```
+
+## Global state and subscriptions
+
+For global state management, we utilize Re-frame subscriptions. They can be likened to React state. To obtain the state, `(rf/sub [])` is employed, and to modify it, `(rf/dispatch [])` is utilized. However, they update components in a similar manner to React states.
+
+```clojure
+(defn view
+  [{:keys [selected-tab]}]
+  (let [collectible-list (rf/sub [:wallet/all-collectibles])
+        on-collectible-press (rn/use-callback
+                              (fn [{:keys [id]}]
+                               (rf/dispatch [:wallet/get-collectible-details id])))]
+    [rn/view {:style style/container}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view {:collectibles         collectible-list
+                                         :on-collectible-press on-collectible-press}])
+       [activity/view]]))
+```
+
+## Regular atoms
+
+In certain instances, components utilized regular atoms; however, they should now be used with `rn/use-ref-atom`
+
+BEFORE:
+```clojure
+(defn view
+  []
+  (let [focused? (atom false)]
+    (fn []
+      (let [on-clear #(reset! status (if @focused? :active :default))
+            on-focus #(reset! focused? true)
+            on-blur  #(reset! focused? false)]))))
+```
+
+NOW:
+```clojure
+(defn view
+  []
+  (let [focused? (rn/use-ref-atom false)
+        on-clear (rn/use-callback #(set-status (if @focused? :active :default)))
+        on-focus (rn/use-callback #(reset! focused? true))
+        on-blur  (rn/use-callback #(reset! focused? false))]))
+```
+
+## Effects
+
+LIFECYCLE:
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let [opacity (reanimated/use-shared-value 0)]
+    (rn/use-mount #(reanimated/animate opacity 1))
+    [rn/view
+     {:style (style/opacity opacity)}]))
+```
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let []
+    (rn/use-unmount #(rn/dispatch [:unmounted]))
+    [rn/view]))
+```
+
+> [!IMPORTANT]
+> Effects shouldn't be used as a reaction to state changes for altering logic.
```

Comment:
Why is this the case? I could see it being useful for some cases, like the composer with markdown editing I guess could react to the input value. Also there's examples in the [react docs](https://react.dev/reference/react/useEffect#examples-connecting) for reacting to state changes. Just wondering why it's emphasised this shouldn't ever be used?